content/en/docs/tasks/traffic-management/tcp-traffic-shifting/test.sh

```diff
@@ -42,6 +42,11 @@ if [ "$GATEWAY_API" == "true" ]; then
     snip_apply_weightbased_tcp_routing_2
     _wait_for_gateway istio-io-tcp-traffic-shifting tcp-echo-gateway
     snip_apply_weightbased_tcp_routing_3
+
+    # Make sure the nc command will work and not fail the cluster
+    # TODO proper wait for things being ready
+    # it seems we had 8 or so exits during the run of 20 so need 8 plus seconds
+    sleep 20s
```

Comment:
I wish we could understand what exactly nc does that corrupts the kubeconfig, and why it never happens when using Istio Gateway/VirtualService instead of gateway-api??